perfkitbenchmarker/network.py

```diff
@@ -102,6 +102,85 @@ def __repr__(self):
     return '%s(%r)' % (self.__class__, self.__dict__)
 
 
+class BaseVPNGW(object):
+  """An object representing the Base VPN GW."""
+  CLOUD = None
+
+  def __init__(self, zone=None, cidr=None):
+    """Initializes the BaseVPNGW.
+
+    Args:
+      zone: The zone in which to create the VPNGW.
+    """
+    self.zone = zone
+    self.cidr = cidr
+    self.require_target_to_init = False  # True if we need taget GW up front (AWS)
+
+  @classmethod
+  def GetVPNGW(cls):
+    """Returns a BaseVPNGW.
+    This method is used instead of directly calling the class's constructor.
+    It creates BaseVPNGW instances and registers them.
+    If a BaseVPNGW object has already been registered, that object
+    will be returned rather than creating a new one. This enables multiple
+    VMs to call this method and all share the same BaseVPN object.
+    """
+    if cls.CLOUD is None:
+      raise errors.Error('VPN GWs should have CLOUD attributes.')
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    if benchmark_spec is None:
+      raise errors.Error('GetVPN called in a thread without a '
+                         'BenchmarkSpec.')
+    with benchmark_spec.vpn_gateways_lock:
+      key = cls.CLOUD
+      if key not in benchmark_spec.vpn_gateways:
+        benchmark_spec.vpn_gateways[key] = cls()
+      return benchmark_spec.vpn_gateways[key]
+
+  def IsTunnelConfigured(self, tunnel_config):
+    """ Returns True if the tunnel_config is complete.
+
+    Args:
+     tunnel_config: The tunnel_config of the tunnel to check.
+    Returns:
+       boolean.
+    """
+    pass
```

Comment:
Should you mark all these methods as abstract and/or raise NotImplementedError?

Reply:
Fixed.